master/internal/experiment/bulk_action.go

```diff
@@ -0,0 +1,505 @@
+package experiment
+
+import (
+	"context"
+
+	"golang.org/x/exp/slices"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/internal/grpcutil"
+	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/rbacv1"
+)
+
+// ExperimentActionResult contains an experiment's ID and associated error.
+type ExperimentActionResult struct {
+	Error error
+	ID    int32
+}
+
+type archiveExperimentOKResult struct {
+	Archived bool
+	ID       int32
+	State    bool
+}
+
+var experimentsAddr = actor.Addr("experiments")
+
+// For each experiment, based on the actor, add an error or non-error to results.
+func loadMultiExperimentActionResults(results []ExperimentActionResult,
+	resps map[*actor.Ref]actor.Message, refExpIDs []int32,
+) []ExperimentActionResult {
+	idx := 0
+	for _, actorResp := range resps {
+		if actorResp == nil {
+			results = append(results, ExperimentActionResult{
+				Error: status.Errorf(codes.Internal, "actorResp nil."),
+				ID:    refExpIDs[idx],
```

Comment:
I am confused: `resps` is a map, and `refExpId` is a lookup table from ref list -> exp id. how would the iteration order on the map match the list indices?

Reply:
ok I pushed an update using `strconv.Atoi(ref.Address().Local())`

Reply:
corrected to ParseInt